docs/literate/src/files/subcell_shock_capturing.jl

```diff
@@ -0,0 +1,285 @@
+# as it was shown by [Pazner (2020)](https://doi.org/10.1016/j.cma.2021.113876) [Section 4.2.2.]
+# that this value guarantees the fulfillment of bounds for a forward-Euler increment.
+
+# Very small non-negative values can be an issue as well. That's why we use an additional
+# correction factor in the calculation of the global bounds,
+# ```math
+# u^{new} \geq \beta * u^{FV}.
+# ```
+# By default, $\beta$ (named `positivity_correction_factor`) is set to `0.1` which works properly
+# in most of the tested setups.
+
+# #### Conservative variables
+# The procedure to enforce global bounds for a conservative variables is as follows:
+# If you want to guarantee non-negativity for the density of the compressible Euler equations,
+# you pass the specific quantity name of the conservative variable.
+using Trixi
+equations = CompressibleEulerEquations2D(1.4)
+
+# The quantity name of the density is `rho` which is how we enable its limiting.
+positivity_variables_cons = ["rho"]
+
+# The quantity names are passed as a vector to allow several quantities.
+# This is used, for instance, if you want to limit the density of two different components using
+# the multicomponent compressible Euler equations.
+equations = CompressibleEulerMulticomponentEquations2D(gammas = (1.4, 1.648),
+                                                       gas_constants = (0.287, 1.578))
+
+# Then, we just pass both quantity names.
+positivity_variables_cons = ["rho1", "rho2"]
+
+# Alternatively, it is possible to all limit all density variables with a general command using
+positivity_variables_cons = ["rho" * string(i) for i in eachcomponent(equations)]
+
+# #### Non-linear variables
+# To allow limitation for all possible non-linear variables, including variables defined
+# on-the-fly, you can directly pass the function that computes the quantity for which you want
+# to enforce positivity. For instance, if you want to enforce non-negativity for the pressure,
+# do as follows.
+positivity_variables_nonlinear = [pressure]
+
+# ### Local bounds
+# Second, Trixi.jl supports the limiting with local bounds for conservative variables. They
+# allow to avoid spurious oscillations within the global bounds and to improve the
+# shock-capturing capabilities of the method. The corresponding numerical admissibility
+# conditions are frequently formulated as local maximum or minimum principles.
+# The local bounds are computed using the maximum and minimum values of all local neighboring
+# nodes. Within this calculation we use the low-order FV solution values for each node.
+
+# As for the limiting with global bounds you are passing the quantity names of the conservative
+# variables you want to limit. So, to limit the density with lower and upper local bounds pass
+# the following.
+local_minmax_variables_cons = ["rho"]
+
+# ## Exemplary simulation
+# How to set up a simulation using the IDP limiting becomes clearer when lokking at a exemplary
```

Comment:
```suggestion
# How to set up a simulation using the IDP limiting becomes clearer when looking at an exemplary
```

Reply:
Oh yes thanks. Done in [cf49c37](https://github.com/trixi-framework/Trixi.jl/pull/1882/commits/cf49c377c88cb64175335e32815c8f1fab0b4389)